controllers/openstackdataplanerole_controller.go

```diff
@@ -184,26 +182,41 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	nodeNames := ""
-	for _, node := range nodes.Items {
-		if node.Spec.Role != instance.Name {
-			err = fmt.Errorf("node %s: node.Role does not match with node.Label", node.Name)
-			return ctrl.Result{}, err
-		}
-		nodeNames = nodeNames + node.Name + ","
-	}
-	r.Log.Info("Role", "Nodes", nodeNames, "Role.Namespace", instance.Namespace, "Role.Name", instance.Name)
+
+	// Validate NodeSpecs
 	err = instance.Validate(nodes.Items)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	roleConfigMap, err := r.GenerateInventory(ctx, instance, nodes.Items)
+	// Ensure IPSets Required for Nodes
+	isReady, err := deployment.EnsureIPSets(ctx, helper, instance, nodes)
```

Comment:
I have general question re dataplane operator as I have not followed it. For each node in a role, does the dataplanerole create a dataplanenode for it? if so, would it make sense to create the ipset as part of the node controller?

Reply:
from https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/docs/inheritance.md it seems the role creates nodes

Reply:
I think I answered it below.